src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -185,4 +187,35 @@ public static int GetTargetCaretPositionForMethod(MethodDeclarationSyntax method
             return lastStatement.GetLocation().SourceSpan.End;
         }
     }
+
+    public static int GetTargetCaretPositionForInsertedMember(SyntaxNode caretTarget)
+    {
+        // Inserted Event declarations are a single line, so move to the end of the line.
+        switch (caretTarget)
+        {
+            case EventFieldDeclarationSyntax:
+                return caretTarget.GetLocation().SourceSpan.End;
```

Comment:
to avoid all this repetition, just have tehse helpers return the node you want to move to the end to.  The caller can do the .GetLocation().SourceSpan.End bit.

Reply:
Changed